internal/pkg/agent/install/uninstall.go

```diff
@@ -59,6 +59,45 @@ func Uninstall(ctx context.Context, cfgFile, topPath, uninstallToken string, log
 		return fmt.Errorf("uninstall must be run from outside the installed path '%s'", topPath)
 	}
 
+	// check if the agent was installed using --unprivileged by checking the file vault for the agent secret (needed on darwin to correctly load the vault)
+	unprivileged, err := checkForUnprivilegedVault(ctx)
```

Comment:
Instead of doing all this here you could just ask the agent if it bootstrapped a fleet server via an RPC call.

You could add something to the agent info response on if it has bootstrapped a fleet server or just look at the components in the state response. https://github.com/elastic/elastic-agent/blob/a4bf7d07bf441e279f4da3d826faf5fe6314edc0/control_v2.proto#L169

Reply:
Using a `client.State` call to determine if we need to notify fleet or are running a fleet-server also adds the assumption that the agent is running during uninstall. Also we would need to load the config in order to build the fleet-server client to make the api request.

Reply:
We have never had the requirement that the Elastic Agent is running to uninstall. I don't think we want to add that requirement for this to work. I would not use RPC to determine this, just like we don't do that to determine if endpoint-security is running.

Reply:
Agree, uninstall needs to work regardless of if the agent is running.

In this specific case, if the RPC fails because the agent isn't running, the local fleet server isn't running either. We would have to treat the RPC failure as non-fatal and make the call to a potentially remote fleet server anyway.

When I made this comment, I was thinking of a problem like https://github.com/elastic/elastic-agent/pull/5864#issuecomment-2457943974 which involved use of `sudo` to interact with an unprivileged agent and was broken in an unexpected way. For uninstall, it has to execute as root regardless of the privilege level of the agent, so I don't think we can have a problem like that here.